editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -350,23 +331,46 @@ export const UiJsxCanvas = React.memo<UiJsxCanvasPropsWithErrorCallback>((props)
   )
   useClearSpyMetadataOnRemount(props.mountCount, props.domWalkerInvalidateCount, metadataContext)
 
+  const elementsToRerenderRef = React.useRef(ElementsToRerenderGLOBAL.current)
+  if (
+    ElementsToRerenderGLOBAL.current === 'rerender-all-elements' ||
+    elementsToRerenderRef.current === 'rerender-all-elements' || // once we get here, we know that `ElementsToRerenderGLOBAL.current` is an array, and `elementsToRerenderRef.current` isn't
+    !pathArraysEqual(ElementsToRerenderGLOBAL.current, elementsToRerenderRef.current) // once we get here, we know that neither `ElementsToRerenderGLOBAL.current` and `elementsToRerenderRef.current` are arrays
```

Comment:
I believe you meant to say that we know they're not equal?

Reply:
just a typo (fixed in https://github.com/concrete-utopia/utopia/pull/2960/commits/0c26f3e1f1fa49aaae14b12ab3a8394165df0b22)